initiate/initiate.go

```diff
@@ -0,0 +1,60 @@
+package initiate
+
+import (
+	"fmt"
+
+	"github.com/prometheus-community/windows_exporter/log"
+	"golang.org/x/sys/windows/svc"
+)
+
+const (
+	serviceName = "windows_exporter"
+)
+
+type windowsExporterService struct {
+	stopCh chan<- bool
+}
+
+func (s *windowsExporterService) Execute(args []string, r <-chan svc.ChangeRequest, changes chan<- svc.Status) (ssec bool, errno uint32) {
+	const cmdsAccepted = svc.AcceptStop | svc.AcceptShutdown
+	changes <- svc.Status{State: svc.StartPending}
+	changes <- svc.Status{State: svc.Running, Accepts: cmdsAccepted}
+loop:
+	for {
+		select {
+		case c := <-r:
+			switch c.Cmd {
+			case svc.Interrogate:
+				changes <- c.CurrentStatus
+			case svc.Stop, svc.Shutdown:
+				log.Debug("Service Stop Received")
```

Comment:
For anyone looking to help get this over the line, Currently when someone hits 'stop' on the windows service we execute up to here fine, but the overall process is not stopping, presumably because this happens out in exporter.go. If you then hit 'Stop' a second time this code does not fire, presumably because the service component here is closed down whilst the process component stays open.

 I know im doing something wrong here but im not sure what.

Reply:
Came down to the definition of my channel being wrong